pkg/policy/mutate/validate_test.go

```diff
@@ -0,0 +1,92 @@
+package mutate
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+
+	kyvernov1 "github.com/kyverno/kyverno/api/kyverno/v1"
+	"gotest.tools/assert"
+)
+
+func Test_Validate_PatchStrategicMerge_Has_Conditional_Anchors(t *testing.T) {
```

Comment:
Something additional I just noticed which completely flew over my head:
Why are you doing all of this in `pkg/policy/mutate`? This whole package is about mutating policies - not actual "normal" objects. IMO this should happen here: https://github.com/kyverno/kyverno/blob/main/pkg/engine/mutation_test.go

Additionally as @chipzoller stated in the issue - schema validation is not executed. Something that is done in the `pkg/engine/mutate`. 

It seems like now with the added changes you are trying to re-implement schema-validation - which already is done in the pre-processing of strategicPreprocessing: https://github.com/kyverno/kyverno/blob/main/pkg/engine/mutate/patch/strategicPreprocessing.go#L47